sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -3741,3 +3745,39 @@ def seal_append_blob(self, **kwargs):
             return self._client.append_blob.seal(**options) # type: ignore
         except HttpResponseError as error:
             process_storage_error(error)
+
+    @distributed_trace
+    def get_container_client(self, **kwargs):
+        # type: (...) -> ContainerClient
+        """Get a client to interact with the blob's parent container.
+
+        The container need not already exist. Defaults to current blob's credentials.
+
+        :keyword credential:
+            This enables you to change credentials when its necessary. The value can be a SAS token string,
+            an instance of a AzureSasCredential from azure.core.credentials, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :returns: A ContainerClient.
+        :rtype: ~azure.storage.blob.ContainerClient
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/blob_samples_containers.py
+                :start-after: [START get_container_client_from_blob_client]
+                :end-before: [END get_container_client_from_blob_client]
+                :language: python
+                :dedent: 8
+                :caption: Get container client from blob object.
+        """
+        from ._container_client import ContainerClient
+        _pipeline = Pipeline(
+            transport=TransportWrapper(self._pipeline._transport), # pylint: disable = protected-access
+            policies=self._pipeline._impl_policies # pylint: disable = protected-access
+        )
+
+        return ContainerClient(
+            "{}://{}".format(self.scheme, self.primary_hostname), container_name=self.container_name,
```

Comment:
you can pass the raw url here and don't need to worry if credential is None

Reply:
raw_url would fail in the case:
If the user wants to change the credential from SAS to AzureSasCredential when getting their container client. It would throw:
```
        if sas_token and isinstance(credential, AzureSasCredential):
            raise ValueError(
                "You cannot use AzureSasCredential when the resource URI also contains a Shared Access Signature.")
```
Therefore we needed this new parameter I added and passing the URI with only scheme and primary hostname

Reply:
why does the user want to pass another credential when get the parent client?

Reply:
I feel this extra flexibility would be nice and doesn't cost us much. This would give the user to change the credential on the fly if they choose to rather than instantiating a whole new container client just because they want to change credentials.

Reply:
while if you put credential option there, that would cause confusion for users who are using shared key or AAD credential, they would think why they can pass another credential here